sdk/containerregistry/container-registry/src/tokenCycler.ts

```diff
@@ -0,0 +1,220 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AccessToken, GetTokenOptions, TokenCredential } from "@azure/core-auth";
+
+function delay<T>(t: number, value?: T): Promise<T | void> {
+  return new Promise((resolve) => setTimeout(() => resolve(value), t));
+}
+/**
+ * A function that gets a promise of an access token and allows providing
+ * options.
+ *
+ * @param options - the options to pass to the underlying token provider
+ */
+export type AccessTokenGetter<T extends GetTokenOptions> = (
```

Comment:
core libraries has been released.  Using a private copy of tokenCycler.ts for now.

Reply:
The plan though is to move even the challenge policy to use the challenge callbacks. When are you expecting to release this change? because if the plan is to release this in a month, you could leverage the targeted approach now. When will core-rest-pipelines go GA again?

Reply:
I need this as part of May release. I think core-rest-pipeline will need to go GA in June, because ACR will GA in June too. We can update the token cycler in core then move to use it before that.

Reply:
Ok that makes sense! Thank you for answering